dbt/adapters/athena/connections.py

```diff
@@ -56,6 +56,7 @@ class AthenaCredentials(Credentials):
     region_name: str
     endpoint_url: Optional[str] = None
     work_group: Optional[str] = None
+    work_group_enforced: Optional[bool] = None
```

Comment:
If we set it to False, we would provide a meaningful value to be returned in is_work_group_output_location_enforced (as you mentioned in the second comment). The intention is to avoid an AWS call if we know the outcome of such AWS call because we provisioned the WorkGroup, but still make the call if the outcome (work_group_enforced value) is not provided.

In this case, it will be backward compatible. Setting it to False would change it. Does that make sense?

To be honest, I can also go with your option as long as it solves the problem. It changes the meaning of the setting I want to introduce to something like, 'if it is set, assume the bucket is enforced and skip checking.' But in my case, I would like to have it set to False, because the WorkGroup is not enforcing the output location, and I still want to skip the AWS call. So I see three values to cover if we want to be backward compatible.

Reply:
If you set by default to False, and you add a check in the if condition to only check for True values, and aws call will still be made, therefore we are covered.
In case of boolean values I prefer to leave always less path possible. In this case default to False allow backward compatibility for the users